integration-cli/docker_cli_build_test.go

```diff
@@ -4694,3 +4694,51 @@ func TestBuildDockerfileOutsideContext(t *testing.T) {
 
 	logDone("build - Dockerfile outside context")
 }
+
+func TestBuildMissingArgs(t *testing.T) {
+	// test to make sure these cmds w/o any args will generate an error
+	// Notice some commands are missing because its ok for them to
+	// not have any args - like: CMD, RUN, ENTRYPOINT
```

Comment:
Actually, all commands don't need an arg today.  I came across an interesting inconsistency.  Today we ban:
```
RUN
```
but allow:
```
RUN []
```
which both end up with the same net result in the resulting image.  The only reason `RUN []` is allowed is because the parser thinks `[]` is an arg, but in reality it eventually gets turned into no args for the `RUN` command. This inconsistency was there for RUN, CMD and ENTRYPOINT - that's why I removed them from the list.

As for making the parser aware of how many args a command needs.... no, I'm not in favor of that.  I really don't like that the parser knows about commands at all. IMO it should be a generic line/word parser and leave understanding the words up to the higher-level code - meaning the builder.  I know we can't get rid of the parser knowing about the commands today (mainly because we have specialized parsers per command) but I don't want to make the situation worse. 
So, I'd prefer not to make this a unit test.